library/core/src/ptr/mut_ptr.rs

```diff
@@ -2302,3 +2316,93 @@ impl<T: ?Sized> PartialOrd for *mut T {
         *self >= *other
     }
 }
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+mod verify {
+    use crate::kani;
+    use kani::PointerGenerator;
+    use core::mem;
+
+    // The proof for a unit type panics as offset_from needs the pointee size > 0
+    #[kani::proof_for_contract(<*mut ()>::offset_from)]
+    #[kani::should_panic]
+    pub fn check_mut_offset_from_unit() {
+        let mut val: () = ();
+        let src_ptr: *mut () = &mut val;
+        let dest_ptr: *mut () = &mut val;
+        unsafe {
+            dest_ptr.offset_from(src_ptr);
+        }
+    }
+
+    // Array size is bound for kani::any_array
+    const ARRAY_LEN: usize = 40;
+    // The array's length is set to a random value, which defines its size.
+    // This is necessary because implementing a dynamic array is not possible.
+    // Size can be changed.
```

Comment:
```suggestion
    // The array's length is set to an arbitrary value, which defines its size.
    // In this case, implementing a dynamic array is not possible, because ...
```
Could you update your comment to answer the "why" question here and incorporate these minor suggestion?